### ISSUE TITLE
Jekyll build fails on Darwin on latest master

### ISSUE THREAD

**Issue by copumpkin** [NixOS/nixpkgs#11741]
Not sure what changed in the past few days, but now (29ab5c3b9c176bef496d33510499622f445783bf) I get:

```
$ nix-build -A jekyll
these derivations will be built:
  /nix/store/j3is4k3gsjm4fxmb3yqr9xjzm7kc3ixh-jekyll-2.5.3.drv
building path(s) ‘/nix/store/m367yxhdjm03vd39v799im50bai6fppf-jekyll-2.5.3’, ‘/nix/store/pxpsjiv32pqgn22jqhcxr2cwgy5hsd2h-jekyll-2.5.3-bundle’
installing
running pre-install script for ffi
Fetching source index from file:/private/var/folders/vv/17b_hycx691699h20v458fnh0000gn/T/nix-build-jekyll-2.5.3.drv-1
..
Gem::Ext::BuildError: ERROR: Failed to build gem native extension.

    /nix/store/lz9i5f8sggjxpgafnb9mp3nzb0pfapxp-ruby-2.1.7-p0/bin/ruby -r ./siteconf20151215-62073-ttkq5x.rb extconf.rb 
checking for main() in -lc... *** extconf.rb failed ***
Could not create Makefile due to some reason, probably lack of necessary
libraries and/or headers.  Check the mkmf.log file for more details.  You may
need configuration options.

Provided configuration options:
    --with-opt-dir
    --without-opt-dir
    --with-opt-include
    --without-opt-include=${opt-dir}/include
    --with-opt-lib
    --without-opt-lib=${opt-dir}/lib
    --with-make-prog
    --without-make-prog
    --srcdir=.
    --curdir
    --ruby=/nix/store/lz9i5f8sggjxpgafnb9mp3nzb0pfapxp-ruby-2.1.7-p0/bin/ruby
    --with-redcloth_scan-dir
    --without-redcloth_scan-dir
    --with-redcloth_scan-include
    --without-redcloth_scan-include=${redcloth_scan-dir}/include
    --with-redcloth_scan-lib
    --without-redcloth_scan-lib=${redcloth_scan-dir}/lib
    --with-clib
    --without-clib
/nix/store/lz9i5f8sggjxpgafnb9mp3nzb0pfapxp-ruby-2.1.7-p0/lib/ruby/2.1.0/mkmf.rb:456:in `try_do': The compiler failed to generate an executable file. (RuntimeError)
You have to install development tools first.
    from /nix/store/lz9i5f8sggjxpgafnb9mp3nzb0pfapxp-ruby-2.1.7-p0/lib/ruby/2.1.0/mkmf.rb:541:in `try_link0'
    from /nix/store/lz9i5f8sggjxpgafnb9mp3nzb0pfapxp-ruby-2.1.7-p0/lib/ruby/2.1.0/mkmf.rb:556:in `try_link'
    from /nix/store/lz9i5f8sggjxpgafnb9mp3nzb0pfapxp-ruby-2.1.7-p0/lib/ruby/2.1.0/mkmf.rb:736:in `try_func'
    from /nix/store/lz9i5f8sggjxpgafnb9mp3nzb0pfapxp-ruby-2.1.7-p0/lib/ruby/2.1.0/mkmf.rb:967:in `block in have_library'
    from /nix/store/lz9i5f8sggjxpgafnb9mp3nzb0pfapxp-ruby-2.1.7-p0/lib/ruby/2.1.0/mkmf.rb:912:in `block in checking_for'
    from /nix/store/lz9i5f8sggjxpgafnb9mp3nzb0pfapxp-ruby-2.1.7-p0/lib/ruby/2.1.0/mkmf.rb:351:in `block (2 levels) in postpone'
    from /nix/store/lz9i5f8sggjxpgafnb9mp3nzb0pfapxp-ruby-2.1.7-p0/lib/ruby/2.1.0/mkmf.rb:321:in `open'
    from /nix/store/lz9i5f8sggjxpgafnb9mp3nzb0pfapxp-ruby-2.1.7-p0/lib/ruby/2.1.0/mkmf.rb:351:in `block in postpone'
    from /nix/store/lz9i5f8sggjxpgafnb9mp3nzb0pfapxp-ruby-2.1.7-p0/lib/ruby/2.1.0/mkmf.rb:321:in `open'
    from /nix/store/lz9i5f8sggjxpgafnb9mp3nzb0pfapxp-ruby-2.1.7-p0/lib/ruby/2.1.0/mkmf.rb:347:in `postpone'
    from /nix/store/lz9i5f8sggjxpgafnb9mp3nzb0pfapxp-ruby-2.1.7-p0/lib/ruby/2.1.0/mkmf.rb:911:in `checking_for'
    from /nix/store/lz9i5f8sggjxpgafnb9mp3nzb0pfapxp-ruby-2.1.7-p0/lib/ruby/2.1.0/mkmf.rb:962:in `have_library'
    from extconf.rb:5:in `<main>'

extconf failed, exit code 1

Gem files will remain installed in /nix/store/m367yxhdjm03vd39v799im50bai6fppf-jekyll-2.5.3/lib/ruby/gems/2.1.7/gems/RedCloth-4.2.9 for inspection.
Results logged to /nix/store/m367yxhdjm03vd39v799im50bai6fppf-jekyll-2.5.3/lib/ruby/gems/2.1.7/extensions/x86_64-darwin-15/2.1.0/RedCloth-4.2.9/gem_make.out
An error occurred while installing RedCloth (4.2.9), and Bundler cannot
continue.
Make sure that `gem install RedCloth -v '4.2.9'` succeeds before bundling.
builder for ‘/nix/store/j3is4k3gsjm4fxmb3yqr9xjzm7kc3ixh-jekyll-2.5.3.drv’ failed with exit code 5
error: build of ‘/nix/store/j3is4k3gsjm4fxmb3yqr9xjzm7kc3ixh-jekyll-2.5.3.drv’ failed
```

Perhaps related to #11685?

**Comment by jagajaga** [comment]
Do you still have such problems?

**Comment by johbo** [comment]
I think this problem is solved, just tried `nix-build -A jekyll` on 501e485e75eae8f300210eac7a90ca3730f82951 and it did succeed after fetching the binaries from the cache on my darwin box.

**Comment by vcunat** [comment]
It looks OK on Hydra.  It's been a year already... http://hydra.nixos.org/job/nixpkgs/trunk/jekyll.x86_64-darwin
